src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -265,7 +265,8 @@ private Task ProcessStreamItem(HubConnectionContext connection, StreamItemMessag
                         for (var parameterPointer = 0; parameterPointer < arguments.Length; parameterPointer++)
                         {
                             if (hubMethodInvocationMessage.Arguments.Length > hubInvocationArgumentPointer &&
-                                descriptor.OriginalParameterTypes[parameterPointer].IsAssignableFrom(hubMethodInvocationMessage.Arguments[hubInvocationArgumentPointer].GetType()))
+                                (hubMethodInvocationMessage.Arguments[hubInvocationArgumentPointer] == null ||
```

Comment:
Need to actually test the fix by having a hub method that accepts a `CancellationToken`!

Just add another test similar to the one you've already added.

Reply:
FYI, I like to remove the fix after making a test and make sure the test actually fails when the fix is removed.

Reply:
Of course! I actually did write the test with `CancellationToken` first, but I remove it at last minute. I forgot that it's the `CancellationToken` that makes `HasSyntheticArguments` true.